pkg/storage/tsdb/bucketindex/markers_bucket_client.go

```diff
@@ -0,0 +1,85 @@
+package bucketindex
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/oklog/ulid"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+type globalMarkersBucket struct {
```

Comment:
It's not clear what "global" refers to. Comment would help. (Perhaps "userMarkersBucket" name too?)